Modules/DataModel/Sources/PocketCastsDataModel/Private/Managers/EpisodeDataManager.swift

```diff
@@ -53,6 +53,33 @@ class EpisodeDataManager {
         loadSingle(query: "SELECT * from \(DataManager.episodeTableName) WHERE \(customWhere)", values: arguments, dbQueue: dbQueue)
     }
 
+    func findPlayedEpisodesBy(uuids: [String], dbQueue: FMDatabaseQueue) -> [Episode] {
+        let list = uuids.map { "'\($0)'" }.joined(separator: ",")
+
+        let query = """
+        SELECT * from \(DataManager.episodeTableName)
+        WHERE uuid IN (\(list))
+        AND playingStatus = 3
+        LIMIT \(uuids.count)
+        """
+
+        var episodes = [Episode]()
+        dbQueue.inDatabase { db in
+            do {
+                let resultSet = try db.executeQuery(query, values: nil)
+                defer { resultSet.close() }
+
+                while resultSet.next() {
+                    let episode = self.createEpisodeFrom(resultSet: resultSet)
```

Comment:
[nit] Do we need to do that? Specifically for the search, we need to know if the episode UUID is returned or not, right?

Doing this additional work is unnecessary but at the same time I think it's fine. Just bringing it up in case you want to make things even more performant.

Reply:
@leandroalonso do you mean returning a set of uuids instead of episodes?

Reply:
Yes!

Reply:
Perfect! I'm on it!